lnwallet/rpcwallet/remote_signer.go

```diff
@@ -91,3 +98,162 @@ type RemoteSignerRequests interface {
 	SignPsbt(ctx context.Context, in *walletrpc.SignPsbtRequest,
 		opts ...grpc.CallOption) (*walletrpc.SignPsbtResponse, error)
 }
+
+// InboundRemoteSigner is an abstraction of the connection to an inbound remote
+// signer. An inbound remote signer is a remote signer that allows the
+// watch-only node to connect to it via an inbound GRPC connection.
+type InboundRemoteSigner struct {
+	// Embedded signrpc.SignerClient and walletrpc.WalletKitClient to
+	// implement the RemoteSigner interface.
+	signrpc.SignerClient
+
+	walletrpc.WalletKitClient
+
+	// The host:port of the remote signer node.
+	rpcHost string
+
+	// The path to the TLS certificate of the remote signer node.
+	tlsCertPath string
+
+	// The path to the macaroon of the remote signer node.
+	macaroonPath string
+
+	// The timeout for the connection to the remote signer node.
+	timeout time.Duration
+}
+
+// NewInboundRemoteSigner creates a new InboundRemoteSigner instance.
+// The function sets up a connection to the remote signer node.
+// The returned function is a cleanup function that should be called to close
+// the connection when the remote signer is no longer needed.
+func NewInboundRemoteSigner(rpcHost string, tlsCertPath string,
```

Comment:
Thanks! I changed so that this takes a passed context, and moved out the actual creation of `context.Background()` further out, so that you can easily add usage of the proper global context in your upcoming PR :).